openxr/examples/vulkan.rs

```diff
@@ -108,26 +143,85 @@ fn main() {
         .api_version(vk_target_version);
 
     unsafe {
-        let vk_instance = xr_instance
-            .create_vulkan_instance(
-                system,
-                std::mem::transmute(vk_entry.static_fn().get_instance_proc_addr),
-                &vk::InstanceCreateInfo::builder().application_info(&vk_app_info) as *const _
-                    as *const _,
+        let vk_instance = if enabled_extensions.khr_vulkan_enable2 {
+            let vk_instance = xr_instance
+                .create_vulkan_instance(
+                    system,
+                    std::mem::transmute(vk_entry.static_fn().get_instance_proc_addr),
+                    &vk::InstanceCreateInfo::builder().application_info(&vk_app_info) as *const _
+                        as *const _,
+                )
+                .expect("XR error creating Vulkan instance")
+                .map_err(vk::Result::from_raw)
+                .expect("Vulkan error creating Vulkan instance");
+            ash::Instance::load(
+                vk_entry.static_fn(),
+                vk::Instance::from_raw(vk_instance as _),
             )
-            .expect("XR error creating Vulkan instance")
-            .map_err(vk::Result::from_raw)
-            .expect("Vulkan error creating Vulkan instance");
-        let vk_instance = ash::Instance::load(
-            vk_entry.static_fn(),
-            vk::Instance::from_raw(vk_instance as _),
-        );
+        } else {
+            let mut buffer = [0_u8; 1000];
+            let mut data_len = 0_u32;
+            let res = (xr_instance
+                .exts()
+                .khr_vulkan_enable
+                .unwrap()
+                .get_vulkan_instance_extensions)(
+                xr_instance.as_raw(),
+                system,
+                buffer.len() as _,
+                &mut data_len as _,
+                &mut buffer as *mut _ as _,
+            );
```

Comment:
This (and the later call for device extensions) should use the proper two-call pattern and a `Vec` buffer to ensure sufficient space is allocated. It's fine to copy the internal helpers to accomplish this (though we might want to consider bundling them into the `sys` crate).

Reply:
I'll remove this code when I add proper support for `khr_vulkan_enable`.